### DISCUSSION TITLE
inet variable

### DISCUSSION THREAD

**Discussion by TwiztedWisard** [Zackptg5/Wireguard-Pi-Hole-Cloudflared-Unbound-DNSCrypt-VPN-Server#6] [Q&A] [answered]
hey man, ive been trying for a couple of days to configure this on a VPS and kept hitting a snag, i think i've located the problem, but would really appreciate your input. 

in the variable inet=$(ip route show default | awk '/default/ {print $5}') I'm assuming that this should return an ip address for example 127.0.0.1. but when running the script on the VPS this variable just returns 'link'

I've tried changing the 'print $#' but no luck.

as ive said previously i am in no way an expert and wanted to know if i was to set this value rather than rely on the cmd would that work or could I use something like inet="$(hostname -I | awk '{print $1}')" instead? (tested this command on the terminal and it produces my inet addr)

Thanks again dude! It's a great looking script :D

**Comment by Zackptg5** [comment, marked as answer]
What distro are you running? inet should be the main network interface, for example:
`zack@DaBeast:~$ ip route show default`
`default via 192.168.1.1 dev eth0`
So inet=eth0

**Comment by TwiztedWisard** [comment]
Thanks for the reply man :)

Tried both Debian 10 and Ubuntu 20.4 but I think your reply answered the problem, I just need to shift the ‘print’ value slightly.

The VPS has a few interfaces extra than what would normally be on a bare metal install and includes extra info so the numbering is slightly off for my install. For ex, in the ipv4 variable it returned a loop back address and in the ipv6 variable it returned my ipv4 address. 

It’s fairly late here so I will try this in the morning and keep you posted :D

**Reply by Zackptg5** [reply]
What's the output when you run the ip route command?

**Comment by Zackptg5** [comment]
Ok. I need to find a better way of setting this variable then so this problem doesn't occur. I'll update the script if I find anything

**Reply by TwiztedWisard** [reply]
Thanks again for the responses man, not sure where you are in this fine world but there appears to be a time difference, im UK :)

the output of ip route show default on my VPS (hostinger) is:

root@twiztedwisard:~# ip route show default
default dev venet0 scope link

as you can see its way off from your results....out of interest what VPS service do you use? im just getting into this so have not locked my self down to any particular one so would be happy to switch.

I'll include my output of ifconfig so you can see their set up in regards to NICs/Interfaces:

root@twiztedwisard:~# ifconfig
lo: flags=73<UP,LOOPBACK,RUNNING>  mtu 65536
        inet 127.0.0.1  netmask 255.0.0.0
        inet6 ::1  prefixlen 128  scopeid 0x10<host>
        loop  txqueuelen 1000  (Local Loopback)
        RX packets 2  bytes 98 (98.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 2  bytes 98 (98.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

venet0: flags=211<UP,BROADCAST,POINTOPOINT,RUNNING,NOARP>  mtu 1500
        inet 127.0.0.1  netmask 255.255.255.255  broadcast 0.0.0.0  destination 127.0.0.1
        inet6 ::2  prefixlen 128  scopeid 0x80<compat,global>
        inet6 REDACTED  prefixlen 64  scopeid 0x0<global>
        unspec 00-00-00-00-00-00-00-00-00-00-00-00-00-00-00-00  txqueuelen 0  (UNSPEC)
        RX packets 7514  bytes 412234 (412.2 KB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 4296  bytes 305181 (305.1 KB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

venet0:0: flags=211<UP,BROADCAST,POINTOPOINT,RUNNING,NOARP>  mtu 1500
        inet REDACTED  netmask 255.255.255.0  broadcast REDACTED destination REDACTED
        unspec 00-00-00-00-00-00-00-00-00-00-00-00-00-00-00-00  txqueuelen 0  (UNSPEC)

root@twiztedwisard:~#

**Reply by Zackptg5** [reply]
Interesting, so you'd want it to be:
`inet=$(ip route show default | awk '/default/ {print $3}')`

I recommend debian 10 over ubuntu because ironically, the packages we need for this are more up to date (especially dnscrypt-proxy)
I use Shock Hosting for my main VPN and also have a few instances on Vultr including a UK based vpn so I can watch Channel4 from the US :D

**Reply by TwiztedWisard** [reply]
Nice man, was looking at vultr or linode but the pricing structue scares me a little on linode even tho it would be perfect for my use case,  i'll check out shock hosting but im guessing its a us based so i may not have luck there. 

currently 21:30 here so i may not get round to trying this out till the morning again. 

if i remember correctly i think i did change the print value to 3 in one of my runs, which makes me wonder if its the way they (hostinger) impliment TUN/TAP devices?

trust me the struggle i had getting TUN/TAP to work on hostinger just for a simple working wireguard install was insane, for example  they told me I had to use sudo.. wait for it...from the root account to set it up! something I (or anyone else tbf) would never have thought of?? and add a start up script something along the lines of mknod /dev/net/tun 10 200, to create the device every time it boots..go figure?

all in all I truly believe this is a problem with the way hostinger have set things up with their wierd VENET interfaces, maybe they are silently trying to stop people ( or make it very difficult) using VPS as VPN because they can't log the data, wouldn't surprise me in the slightest but there are also probably a million reasons why im wrong there.

I'm not giving up on your script tho dude - i've read through it 100's of times now and even being as inexperinced as I am i can tell this is very cleverly written and implimented compared to others like it on here so thank you again! :D 

PS: Channel 4 is one of the best....BBC3 used to be quite good too! check out their iplayers ;) i do the same to get FOX as i am a MASSIVE Archer fan! but seems to take ages for us to get the new eps!

**Reply by Zackptg5** [reply]
Oh dang, I'd definitely switch to a different hosting provider if they don't give you a normal root setup. I haven't tried linode myself but vultr has great speeds and they still have the[ $50 credit for sign up promo going on](https://www.vultr.com/promo/try50). Like with other large hosting providers though, their ips are often blacklisted by google so searx won't work with google on there. 
That's where a smaller hosting provider like shock can come in. I've been running a searx instance with google on it for over a month now and it's been great. The management interface is garbage compared to vultr, but they have gig speeds and great support: https://shockhosting.net

**Comment by TwiztedWisard** [comment]
Cheers for the advice man I’ll def check out shock! Although $50 credit is tempting...
